packages/grid/_modules_/grid/GridComponent.tsx

```diff
@@ -93,118 +86,71 @@ export const GridComponent = React.forwardRef<HTMLDivElement, GridComponentProps
     useColumnResize(columnsHeaderRef, apiRef);
     usePagination(apiRef);
 
-    const customComponents = useComponents(props.components, apiRef, rootContainerRef);
+    const components = useComponents(props.components, apiRef, rootContainerRef);
     useStateProp(apiRef, props.state);
+    useRenderInfoLog(apiRef, logger);
 
-    const visibleColumnsLength = useGridSelector(apiRef, visibleColumnsLengthSelector);
-
-    // TODO move that to renderCtx
-    const getTotalHeight = React.useCallback(
-      (size) =>
-        getCurryTotalHeight(
-          gridState.options,
-          gridState.containerSizes,
-          headerRef,
-          footerRef,
-        )(size),
-      [gridState.options, gridState.containerSizes],
-    );
-
-    if (gridState.rendering.renderContext != null) {
-      const {
-        page,
-        firstColIdx,
-        lastColIdx,
-        firstRowIdx,
-        lastRowIdx,
-      } = gridState.rendering.renderContext!;
-      logger.info(
-        `Rendering, page: ${page}, col: ${firstColIdx}-${lastColIdx}, row: ${firstRowIdx}-${lastRowIdx}`,
-      );
-    }
-
+    const showNoRowsOverlay = !props.loading && gridState.rows.totalRowCount === 0;
     return (
-      <AutoSizer onResize={onResize} nonce={props.nonce}>
-        {(size: any) => (
-          <GridRoot
-            ref={handleRef}
-            className={props.className}
-            style={{ width: size.width, height: getTotalHeight(size) }}
-            role="grid"
-            aria-colcount={visibleColumnsLength}
-            aria-rowcount={gridState.rows.totalRowCount}
-            tabIndex={0}
-            aria-label={apiRef!.current.getLocaleText('rootGridLabel')}
-            aria-multiselectable={!gridState.options.disableMultipleSelection}
-          >
-            <ApiContext.Provider value={apiRef}>
+      <ApiContext.Provider value={apiRef}>
+        <AutoSizer onResize={onResize} nonce={props.nonce}>
+          {(size: any) => (
+            <GridRoot
+              ref={handleRef}
+              className={props.className}
+              size={size}
+              header={headerRef}
+              footer={footerRef}
+            >
               <ErrorBoundary
                 hasError={errorState != null}
                 componentProps={errorState}
                 api={apiRef!}
                 logger={logger}
                 render={(errorProps) => (
-                  <div className="MuiDataGrid-mainGridContainer">
-                    {customComponents.renderError(errorProps)}
-                  </div>
+                  <GridMainContainer>
+                    <components.Error {...errorProps} {...props.componentsProps?.ErrorOverlay} />
```

Comment:
I believe that @mnajdova has been pushing for a lower case here. I don't remember why, if I had to guess it would be because it's applied to a React element at this point, not a Component.

```suggestion
                    <components.Error {...errorProps} {...props.componentsProps?.errorOverlay} />
```

Reply:
it's probably because it is an object and not a component or a type

Reply:
Exactly, anything that is not used as a component, does not need to be uppercase

Reply:
Depends if we want to match the component slots name 
As it's an object we can put it in camelCase 🤔

Reply:
We should match the names, it's just that it is not starting with uppercase letter. I'd say we should use uppercase first letter only if it really brings value, which does with components. With componentsProps, it does not, so lowercase first letter should be preffered in my opinion.